src/headless/converse-muc.js

```diff
@@ -507,6 +507,42 @@ converse.plugins.add('converse-muc', {
                     }, null, 'message', 'groupchat', null, room_jid,
                     {'matchBareFromJid': true}
                 );
+                this.muc_notifications_handler = _converse.connection.addHandler(stanza => {
+                    const item = sizzle(`x[xmlns="${Strophe.NS.MUC_USER}"]`, stanza).pop();
+
+                    if (item) {
+                        const from = stanza.getAttribute("from");
+                        const type = stanza.getAttribute("type");
+                        const affiliation = item.getAttribute('affiliation');
+                        const jid = item.getAttribute('jid');
+
+                        const data = {
+                          'from': from,
+                          'nick': Strophe.getNodeFromJid(jid),
+                          'type': type,
+                          'states': [],
+                          'show': type == 'unavailable' ? 'offline' : 'online',
+                          'affiliation': affiliation,
+                          'role': item.getAttribute('role'),
+                          'jid': jid
+                        }
+
+                        const attributes = Object.assign(data, {
+                            'jid': Strophe.getBareJidFromJid(data.jid),
+                            'resource': Strophe.getResourceFromJid(data.jid)
+                        });
+                        const occupant = this.occupants.findOccupant({'jid': data.jid});
+
+                        if (occupant) {
+                          occupant.save(attributes);
+                        } else {
+                          this.occupants.create(attributes);
+                        }
+                    }
+                    return true;
+
+                }, null, 'message', null, null, room_jid,
+                {'matchBareFromJid': true });
```

Comment:
`matchBareFromJid` doesn't make sense in this case.

Reply:
ok